cmd/ci-secret-bootstrap/main_test.go

```diff
@@ -2633,3 +2634,435 @@ func (f *fakeVaultClient) ListKVRecursively(prefix string) ([]string, error) {
+							{
+								Cluster:   "cluster-1",
+								Namespace: "namespace-1",
+								Name:      "prod-secret-1",
+							},
+						},
+					},
+				},
+			},
+			secretGetters: map[string]coreclientset.SecretsGetter{
+				"cluster-1": fake.NewSimpleClientset().CoreV1(),
+				"cluster-2": fake.NewSimpleClientset().CoreV1(),
+				"cluster-3": fake.NewSimpleClientset().CoreV1(),
+			},
+			vaultObjects: map[string][]vaultObject{
+				"item-name-1": {
+					{
+						field: "field-name-1",
+						data:  []byte("secret-data"),
+					},
+				},
+				"user-item-1": {
+					{
+						field: "some-data-key",
+						data:  []byte("a-secret"),
+					},
+					{
+						field: "secretsync/target-namespace",
+						data:  []byte("namespace-1"),
+					},
+					{
+						field: "secretsync/target-name",
+						data:  []byte("some-name"),
+					},
+					{
+						field: "secretsync/target-clusters",
+						data:  []byte("cluster-1,cluster-3"),
+					},
+				},
+			},
+			expectedSecrets: map[string][]coreapi.Secret{
+				"cluster-1": {
+					{
+						ObjectMeta: metav1.ObjectMeta{Name: "prod-secret-1", Namespace: "namespace-1", Labels: map[string]string{"dptp.openshift.io/requester": "ci-secret-bootstrap"}},
+						Data:       map[string][]byte{"key-name-1": []byte("secret-data")},
+						Type:       coreapi.SecretTypeOpaque,
+					},
+					{
+						ObjectMeta: metav1.ObjectMeta{Name: "some-name", Namespace: "namespace-1", Labels: map[string]string{"dptp.openshift.io/requester": "ci-secret-bootstrap"}},
+						Data: map[string][]byte{
+							"some-data-key":                []byte("a-secret"),
+							"secretsync-vault-source-path": []byte("secret/user-item-1"),
+						},
+						Type: coreapi.SecretTypeOpaque,
+					},
+				},
+				"cluster-3": {
+					{
+						ObjectMeta: metav1.ObjectMeta{Name: "some-name", Namespace: "namespace-1", Labels: map[string]string{"dptp.openshift.io/requester": "ci-secret-bootstrap"}},
+						Data: map[string][]byte{
+							"some-data-key":                []byte("a-secret"),
+							"secretsync-vault-source-path": []byte("secret/user-item-1"),
+						},
+						Type: coreapi.SecretTypeOpaque,
+					},
+				},
+			},
+		},
+		{
+			id: "sad case",
+			config: secretbootstrap.Config{
+				Secrets: []secretbootstrap.SecretConfig{
+					{
+						From: map[string]secretbootstrap.BitWardenContext{
+							"key-name-1": {
+								BWItem: "item-name-1",
+								Field:  "field-name-1",
+							},
+						},
+						To: []secretbootstrap.SecretContext{
+							{
+								Cluster:   "cluster-1",
+								Namespace: "namespace-1",
+								Name:      "prod-secret-1",
+							},
+						},
+					},
+				},
+			},
+			secretGetters: map[string]coreclientset.SecretsGetter{
+				"cluster-1": fake.NewSimpleClientset().CoreV1(),
+			},
+			vaultObjects: map[string][]vaultObject{
+				"no-one-wants-me": {
+					{
+						field: "field-name-1",
+						data:  []byte("secret-data"),
+					},
+				},
+			},
+			expectedErrors: true,
```

Comment:
Please do a better error check (error message, error substring). If we fail for an unrelated reason the test will incorrectly succeed.

Reply:
I tried to compare the error message, but vault returns some multi-line error messages and I was unable to compare them. I will check this out.

Reply:
Hrm - I almost disagree here. If the type of error is important to the caller, the Vault functions should return errors that can be inspected with `errors.Is()`. Unwrapping an error to get at internal details like the message is an anti-pattern and opens the otherwise black-box approach. If we care specifically about the type, return typed errors and check with `errors.Is()`.

Reply:
It is important to check that we got the error we expected and not any error. It doesn't really matter to me how exactly we do the checking (all of err message equality, err message substring, errors.Is/As is fine). 

We had plenty of tests in test-infra that succeeded because they did such a bool check but got an entirely different error because their test setup become invalid over time. Doing `expectErr: bool` style tests means they can very quickly and unnoticed rot under you and pass when they shouldn't.

Reply:
I will try to check the error message, but it will be only a string comparison.